### PR TITLE
update cmake

### PR DESCRIPTION
**Pull request by maitbot**
Some minor refactoring of Codec2 build to better use CMake
features to improve readability ans maintainability.

About this update cmake patch to codec2:

- Follow the convention of cmake_minimum_required() being the first CMake command.
  (Moved CMAKE_OSX_DEPLOYMENT_TARGET definition down next to another OSX setting.)

- Update cmake_minimum_required to 3.13.
  (Since that is where POLICY CMP0079 is introduced. We want to follow NEW usage,
   so that seems best. The cmake_policy() handling can be removed.)

- Use (new since CMake 3.0) project() features.
  This is the best single-point definition of project version information.
  (This handles the CODEC2_VERSION* variables for us, so the other definitions
   can be removed.)

- Better DATE_RESULT setting.
  Using CMake's string(TIMESTAMP... function does this in 1 line rather than 5.
  Also, CMake honors the SOURCE_DATE_EPOCH environment variable needed to support
  reproducible builds.
  https://reproducible-builds.org/specs/source-date-epoch/

- Move the SOVERSION setting next to where it is used in the library target definition.
  (Moved from CMakeLists.txt to src/CMakeLists.txt)

- Handle math library via codec2 library target and executable target definitions.
  No need for CODEC2_REQUIRED_LIBRARIES everywhere.

- Pass along the lpcnet_freedv.h header as a codec2 library header if needed.
  (Any other lpcnet headers stay private.)

73 de aa4hs,
-Maitland

### CODE REVIEW

**Comment by kantooon** [comment]
Thanks Maitland for doing this. I have a question for you if you don't mind: do you intend to backport this Codec2 version for usage in gnuradio 3.8 on Bullseye or will it be exclusively gr 3.10?
Adrian

**Comment by drowe67** [comment]
Thanks @maitbot looks OK to me :slightly_smiling_face: 

 @hobbes1069 - do you have any comments?

**Comment by hobbes1069** [comment]
Nothing jumps out at me as problematic. I think most of this was written during circa CMake 2.4ish so I'm sure a lot has changed. We should of course make sure it builds for all supported configurations.

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +42,7 @@ endif()
 if(BUILD_OSX_UNIVERSAL)
 set(CMAKE_OSX_ARCHITECTURES "x86_64;arm64")
 endif(BUILD_OSX_UNIVERSAL)
+set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X deployment version")
 
 #
 # Find the git hash if this is a working copy.
```
Comment (tmiw) [inline CMakeLists.txt:48]: We should probably use `git rev-parse --short` to get the commit hash. `git describe` gives me something like `v1.0.0-428-g160866fc` for current master.

However, we can do another PR for that particular change if preferred.

Comment (drowe67) [reply CMakeLists.txt:48]: @tmiw - would you like to try that change in this PR?  I'm happy to wait for you to try that...

Comment (tmiw) [reply CMakeLists.txt:48]: Done :)

Comment (drowe67) [reply CMakeLists.txt:48]: Thank you @tmiw :slightly_smiling_face:
